apps/rehabstod/src/pages/LUCertificates/LUCertificates.tsx

```diff
@@ -0,0 +1,110 @@
+import { getLUCertificatesColumnInfo } from './utils/getLUCertificatesColumnsInfo'
+import { TableInfoMessage } from '../../components/Table/TableInfoMessage'
+import { TableBody } from '../../components/Table/tableBody/TableBody'
+import { getLUCertificatesTableCell, getLUCertificatesTableValue } from './utils/luCertificatesTableValueFormatter'
+import { isUserDoctor } from '../../utils/isUserDoctor'
+import { filterTableColumns } from '../../components/Table/utils/filterTableColumns'
+import { TableInfo } from '../../components/Table/TableInfo'
+import { ModifyLUCertificatesTableColumns } from './ModifyLUCertificatesTableColumns'
+import { PrintTableBody } from '../../components/Table/tableBody/PrintableTableBody'
+import { updateShowPersonalInformation } from '../../store/slices/settings.slice'
+
+export function LUCertificates() {
+  const { isLoading: userLoading, data: user } = useGetUserQuery()
+  const [triggerGetLUCertificates, { isLoading: isContentLoading, data: luCertificatesInfo, error }] = useLazyGetLUCertificatesQuery()
+  const allColumns = useAppSelector(allLuCertificatesColumns)
+  const [tableState, setTableState] = useState<{ sortColumn: string; ascending: boolean }>({
+    sortColumn: LUCertificatesColumn.Signeringsdatum,
+    ascending: false,
+  })
+  const { hasAppliedFilters } = useAppSelector((state) => state.luCertificates)
+  const { showPersonalInformation } = useAppSelector((state) => state.settings)
+
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  const isDoctor = user ? isUserDoctor(user) : false
+  const visibleColumns = filterTableColumns(allColumns, isDoctor, showPersonalInformation, true)
+
+  const TABLE_NAME = 'läkarutlåtanden'
+
+  const navigateToPatient = (id: string) => {
+    navigate(`/pagaende-sjukfall/${id}`)
+  }
+
+  useEffect(
+    () => () => {
+      dispatch(reset())
+    },
+    [dispatch]
+  )
+
+  return (
+    <TableLayout
+      printable
+      tableName={TABLE_NAME}
+      isUserLoading={userLoading}
+      user={user}
+      heading={<TableHeadingForUnit tableName={TABLE_NAME} suffix="senaste tre åren" user={user} />}
+      filters={<LUCertificatesFilters onSearch={(request) => triggerGetLUCertificates(request)} />}
+      tableInfo={
+        <TableInfo
+          listLength={luCertificatesInfo ? luCertificatesInfo.certificates.length : 0}
+          totalNumber={luCertificatesInfo ? luCertificatesInfo.certificates.length : 0} // TODO: this should come from backend
+          showPersonalInformation={showPersonalInformation} // TODO: should be shared with sickleaves
+          onShowPersonalInformationChange={(checked) => dispatch(updateShowPersonalInformation(checked))}
+        />
+      }
+      modifyTableColumns={<ModifyLUCertificatesTableColumns />}
+      error={error}
+      unansweredCommunicationError={!!luCertificatesInfo?.questionAndAnswersError}
+    >
```

Comment:
For a layout component I would rather see it only accepting components, user, error, printable should be handled by the component's responsible for those things. Instead of error I would expect for example `fetchError={<GetTableContentError tableName={TABLE_NAME} error={error} />}`

Reply:
What are the benefits of the component only accepting components?

Reply:
It's to help avoid mixing logic with layout

Reply:
Creating the hook removed some attributes but I think the ones that are there now should stay since they toggle components which are common for all tables.